### DISCUSSION TITLE
"Failed to create CUDAExecutionProvider" onnxruntime error on Google Colab

### DISCUSSION THREAD

**Discussion by hkhare42** [roboflow/supervision#1402] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

Really appreciate this library and ecosystem!

I'm trying to run the "time_in_zone" example on Google Colab (CUDAExecutionProvider). I pip installed the requirements.txt and in the next step, also pip installed inference-gpu library. 

!nvidia-smi gives me the following showing that CUDA is available. 
![image](https://github.com/user-attachments/assets/6f4aed57-6888-4ccf-a378-02053a3ab7cd)

On running the inference_file_example.py, I get the following error:
`UserWarning: Specified provider 'OpenVINOExecutionProvider' is not in available provider names.Available providers: 'TensorrtExecutionProvider, CUDAExecutionProvider, CPUExecutionProvider'
2024-07-22 18:17:36.983840877 [W:onnxruntime:Default, onnxruntime_pybind_state.cc:640 CreateExecutionProviderInstance] Failed to create CUDAExecutionProvider. Please reference https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#requirements to ensure all dependencies are met.`

I tried going through the referenced link in the error and going over forums but wasn't able to resolve this. What additional step is needed to get CUDAExecutionProvider to work?

**Comment by hkhare42** [comment]
Also sharing links to external threads that discuss similar issues: 
https://github.com/roboflow/inference/issues/324
https://github.com/microsoft/onnxruntime/issues/11092
https://stackoverflow.com/questions/75267445/why-does-onnxruntime-fail-to-create-cudaexecutionprovider-in-linuxubuntu-20

**Comment by onuralpszr** [comment]
@hkhare42  hello o/ let me  convert this issue to discussion and also could you try this codes I made small collab for you test and see are you in GPU 

https://colab.research.google.com/drive/1Ltv7tKae7SOkM7bOjwPQ0x3kYMik5XkE?usp=sharing

**Reply by hkhare42** [reply]
I do get GPU on running the shared colab notebook

![image](https://github.com/user-attachments/assets/89e492e7-ad4a-423f-b425-3a1bdb350fba)

**Reply by onuralpszr** [reply]
@hkhare42  okay great could you try with your own collab code together and see what is happening or you can share minimal collab code If you like

**Reply by hkhare42** [reply]
I'm trying to run the time_in_zone example. (https://github.com/roboflow/supervision/tree/develop/examples/time_in_zone)

My code is essentially this right now:
![image](https://github.com/user-attachments/assets/b3db9ebc-3d53-4dec-ae04-bb2a5e8877ad)

On running the inference_file_example.py, I get the following message and it defaults to CPU:
`UserWarning: Specified provider 'OpenVINOExecutionProvider' is not in available provider names.Available providers: 'TensorrtExecutionProvider, CUDAExecutionProvider, CPUExecutionProvider' 2024-07-22 18:17:36.983840877 [W:onnxruntime:Default, onnxruntime_pybind_state.cc:640 CreateExecutionProviderInstance] Failed to create CUDAExecutionProvider. Please reference https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#requirements to ensure all dependencies are met.`

**Reply by hkhare42** [reply]
Would you recommend pip installing onnxruntime-gpu again and then adding the below two lines at top in inference_file_example.py?

```
import torch
from onnxruntime import *
```

**Reply by hkhare42** [reply]
[colab_tiz_gpu_test.pdf](https://github.com/user-attachments/files/16374361/colab_tiz_gpu_test.pdf)
[test_tiz_colab_library_install_log.txt](https://github.com/user-attachments/files/16374365/test_tiz_colab_library_install_log.txt)

Let me know if you were able to reproduce the example?

**Reply by onuralpszr** [reply]
@hkhare42  could you send as "code" please or collab share link ? It will make my life a lot more easier please

**Reply by hkhare42** [reply]
https://colab.research.google.com/drive/1SVij8glL3Csi80uQfoQl6dR7-lKZ1AQH?usp=sharing

**Reply by hkhare42** [reply]
@onuralpszr let me know if you need anything else. shared the code notebook above.

**Reply by hkhare42** [reply]
Hey @onuralpszr Any suggestions on what I can try out?

**Reply by onuralpszr** [reply]
> Hey @onuralpszr Any suggestions on what I can try out?

Oh,sorry I forgat that about this link let me check quickly

**Reply by onuralpszr** [reply]
> Hey @onuralpszr Any suggestions on what I can try out?

Do installation,next do manual install and make import and then run your code but put your code in collab instead of get from drive way so it can keep imports same and let me know how is it went

**Reply by hkhare42** [reply]
So the "Failed to create CUDAExecutionProvider" error is no longer popping up. So that's a positive. But it seems to be taking same time per iteration as it was when defaulting to the CPUExecutionProvider. Why do you think that's happening? Is that expected? Please refer to the notebook link above.

**Reply by onuralpszr** [reply]
> So the "Failed to create CUDAExecutionProvider" error is no longer popping up. So that's a positive. But it seems to be taking same time per iteration as it was when defaulting to the CPUExecutionProvider. Why do you think that's happening? Is that expected? Please refer to the notebook link above.

After import all could you check is it gpu or cpu like in the collab I made in your collab ?

**Reply by onuralpszr** [reply]
![image](https://github.com/user-attachments/assets/0cbe571f-f5f5-4dac-b0c7-63fa5ca550a8)

Like this

**Reply by hkhare42** [reply]
It says 'GPU'
